File: app/src/main/java/com/example/campusview/api/ApiService.java
package com.example.campusview.api;

import com.example.campusview.model.BookingCreateRequest;
import com.example.campusview.model.Classroom;
import com.example.campusview.model.Resource;
import com.example.campusview.model.ScenicSpotResponse;
import com.example.campusview.model.ScenicSpotResponse.ScenicSpot;
import com.example.campusview.model.NavigationResponse;
import com.example.campusview.model.BookingDto;
import com.example.campusview.model.LoginRequest;
import com.example.campusview.model.LoginResponse;
import com.example.campusview.model.RegisterRequest;
import com.example.campusview.model.RegisterResponse;
import com.example.campusview.model.UserUpdateDto;
import okhttp3.MultipartBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.*;
import java.util.List;

public interface ApiService {
    // 认证相关接口
    @POST("/api/auth/login")
    Call<LoginResponse> login(@Body LoginRequest loginRequest);

    @POST("/api/auth/register")
    Call<RegisterResponse> register(@Body RegisterRequest registerRequest);

    // 资源相关接口
    @GET("/resource")
    Call<List<Resource>> getAllResources();

    @GET("/resource/other")
    Call<List<Resource>> getNonClassroomResources();

    @GET("/resource/{id}")
    Call<Resource> getResource(@Path("id") Long id);

    @GET("/resource/available")
    Call<List<Resource>> getAvailableResources();

    @GET("/resource/type/{type}")
    Call<List<Resource>> getResourcesByType(@Path("type") String type);

    @GET("/resource/available/type/{type}")
    Call<List<Resource>> getAvailableResourcesByType(@Path("type") String type);

    @GET("/resource/available/timeslot")
    Call<List<Resource>> getAvailableResourcesForTimeSlot(
            @Query("startTime") String startTime,
            @Query("endTime") String endTime
    );

    @GET("/api/booking/my")
    Call<List<BookingDto>> getMyBookings();

    @GET("/api/booking/resource/{resourceId}")
    Call<List<BookingDto>> getBookingsByResource(@Path("resourceId") Long resourceId);

    @POST("/api/booking")
    Call<BookingDto> createBooking(@Body BookingCreateRequest request);

    @DELETE("/api/booking/{bookingId}")
    Call<Void> cancelBooking(@Path("bookingId") Long bookingId);

    @PUT("/api/booking/{bookingId}/status")
    Call<BookingDto> updateBookingStatus(@Path("bookingId") Long bookingId, @Query("status") String status);

    // 管理员相关接口
    @GET("/api/admin/bookings")
    Call<List<BookingDto>> getAllBookings();

    @PUT("/api/admin/bookings/{id}/approve")
    Call<BookingDto> approveBooking(@Path("id") Long bookingId);

    @PUT("/api/admin/bookings/{id}/reject")
    Call<BookingDto> rejectBooking(@Path("id") Long bookingId);

    // 用户权限相关接口
    @GET("/api/user/roles")
    Call<List<String>> getUserRoles();

    @GET("/api/user/permissions")
    Call<List<String>> getUserPermissions();

    // 景点识别相关接口
    @Multipart
    @POST("/api/scenic/recognize")
    Call<ScenicSpotResponse> recognizeScenicSpot(@Part MultipartBody.Part image);

    // 导航相关接口
    @GET("/api/navigation/route")
    Call<NavigationResponse> getNavigationRoute(
            @Query("startLat") double startLat,
            @Query("startLng") double startLng,
            @Query("endLat") double endLat,
            @Query("endLng") double endLng
    );

    // 用户信息相关接口
    @GET("/api/users/{userId}")
    Call<LoginResponse> getUserInfo(@Path("userId") Long userId);

    @PUT("/api/users/{userId}")
    Call<LoginResponse> updateUser(@Path("userId") Long userId, @Body UserUpdateDto updateDto);

    @Multipart
    @POST("/api/users/{userId}/avatar")
    Call<LoginResponse> uploadAvatar(
            @Path("userId") Long userId,
            @Part MultipartBody.Part avatar
    );
} 

File: CampusView/app/src/main/java/com/example/campusview/api/ApiService.kt
package com.example.campusview.api

import com.example.campusview.model.*
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    // 用户认证相关
    @POST("api/auth/login")
    suspend fun login(@Body loginRequest: LoginRequest): Response<LoginResponse>

    @POST("api/auth/register")
    suspend fun register(@Body registerRequest: RegisterRequest): Response<RegisterResponse>

    // 教室相关
    @GET("api/classrooms")
    suspend fun getAllClassrooms(): Response<List<ClassroomDto>>

    @GET("api/classrooms/building/{buildingType}")
    suspend fun getClassroomsByBuilding(@Path("buildingType") buildingType: String): Response<List<ClassroomDto>>

    @GET("api/classrooms/available")
    suspend fun getAvailableClassrooms(): Response<List<ClassroomDto>>

    // 资源相关
    @GET("api/resources")
    suspend fun getAllResources(): Response<List<ResourceDto>>

    @GET("api/resources/type/{type}")
    suspend fun getResourcesByType(@Path("type") type: String): Response<List<ResourceDto>>

    @GET("api/resources/available")
    suspend fun getAvailableResources(): Response<List<ResourceDto>>

    @GET("api/resources/available/timeslot")
    suspend fun getAvailableResourcesForTimeSlot(
        @Query("startTime") startTime: String,
        @Query("endTime") endTime: String
    ): Response<List<ResourceDto>>

    // 预约记录相关
    @GET("api/bookings/my")
    suspend fun getMyBookings(
        @Query("page") page: Int,
        @Query("pageSize") pageSize: Int = 20
    ): Response<List<BookingRecordDto>>

    @POST("api/bookings")
    suspend fun createBooking(@Body bookingRequest: CreateBookingRequest): Response<BookingRecordDto>

    @DELETE("api/bookings/{id}")
    suspend fun cancelBooking(@Path("id") id: Long): Response<Unit>

    // 景点相关
    @GET("api/spots")
    suspend fun getAllSpots(): Response<List<ScenicSpotDto>>

    @GET("api/spots/search")
    suspend fun searchSpots(@Query("keyword") keyword: String): Response<List<ScenicSpotDto>>

    @POST("api/spots/recognize")
    suspend fun recognizeImage(@Body request: ImageRecognitionRequest): Response<ImageRecognitionResponse>

    // 用户预约记录
    @GET("api/bookings/user/{userId}")
    suspend fun getUserBookings(@Path("userId") userId: Long): Response<List<BookingRecordDto>>
} 